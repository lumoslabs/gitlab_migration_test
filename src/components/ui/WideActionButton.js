import React from 'react';
import { css, StyleSheet } from 'aphrodite';
import { RoundedButton } from './buttons';
import { analyticsTrack } from '../segment/analytics';
import commonStyles from '../styles/commonStyles';

const WideActionButton = (props) => {
  const extendStyles = props.extendStyles || []

  return (
    <div className={ css(commonStyles.flexJustifyCenter) }>
      <RoundedButton
        styles = { [commonStyles.action, styles.wideAction, ...extendStyles] }
        value = { props.buttonText }
        onClick = {e => {
          analyticsTrack('button_click', {
            id: props.id,
            current_page: props.currentPage,
            message: props.buttonText,
          });
          props.onClick(e);
          e.preventDefault();
        }}
        disabled = { props.disabled }
      />
    </div>
  );
};

const styles = StyleSheet.create({
  wideAction: {
    width: 327,
    height: 56
  },
});

export default WideActionButton;
