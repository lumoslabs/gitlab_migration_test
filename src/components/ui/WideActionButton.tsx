import React from 'react'
import { css, StyleSheet } from 'aphrodite'
import Button from '@components/ui/Button'
import commonStyles from '@styles/commonStyles'
import useAmplitude from '@hooks/useAmplitude'

export interface IWideActionButtonProps {
  extendStyles?: any;
  buttonText: string;
  onClick?(e: React.MouseEvent<any>): any;
  eventData?: Record<string, any>
}

const WideActionButton = (props: IWideActionButtonProps): JSX.Element => {
  const { buttonText, onClick, extendStyles, eventData } = props
  const track = useAmplitude()

  return (
    <div className={css(commonStyles.flexJustifyCenter)}>
      <Button
        buttonStyles={[styles.wideAction, extendStyles]}
        text={buttonText}
        onClick={e => {
          track('button_click', eventData)
          onClick(e)
          e.preventDefault()
        }}
      />
    </div>
  )
}

const styles = StyleSheet.create({
  wideAction: {
    width: 327,
    height: 56
  }
})

export default WideActionButton
