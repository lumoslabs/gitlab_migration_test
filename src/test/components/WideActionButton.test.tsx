import React from 'react'
import { StyleSheet } from 'aphrodite'
import WideActionButton from '@components/ui/WideActionButton'
import { render } from '@testing-library/react'

let currentPage = window.location.href
const handleClick = () => { currentPage = '/' }
const styles = StyleSheet.create({
  nextButton: {
    height: '8.75vmin',
    width: '31.75vmin',
    fontFamily: 'MuseoSans700',
    fontSize: '3.625vmin',
    fontWeight: 700,
    margin: '3.33vh',
    padding: 0
  }
})

it('matches snapshot', () => {
  const { asFragment } = render(
    <WideActionButton
      extendStyles={styles.nextButton}
      buttonText='WideActionButton'
      onClick={handleClick}
      id='WideActionButton'
      currentPage={currentPage}

   />, {}
  )
  expect(asFragment()).toMatchSnapshot()
})
