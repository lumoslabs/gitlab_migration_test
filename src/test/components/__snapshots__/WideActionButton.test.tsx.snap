// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexJustifyCenter_jfsfb5"
  >
    <button
      class="button_1xm70sk-o_O-wideAction_4zoj1f-o_O-nextButton_14syk2g"
    >
      WideActionButton
    </button>
  </div>
</DocumentFragment>
`;
